crates/inspector/src/lib.rs

```diff
@@ -4,6 +4,8 @@
 //! Implementing inspection is optional and it does not effect the core execution.
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![cfg_attr(not(feature = "std"), no_std)]
+// Allow unused serde dependency as it's used only when the serde feature is enabled
+#![allow(unused_extern_crates)]
```

Comment:
```suggestion
// Allow unused serde dependency as it's used only when the serde feature is enabled
#![allow(unused_extern_crates)]
```

I am not sure why this is needed, all other crates has serde as a feature. Can you remove it and push it without, it CI should fail if there is a problem

Reply:
let me try this and hope CI doesn't fail for this 🤞